<font face ="MiSans" ><br/>

### Trim

```bash
#!/bin/bash
#SBATCH -J trim
#SBATCH -p cn-long
#SBATCH -N 1
#SBATCH -c 60
#SBATCH -o /data01nfs/user/songq/log/trim.out
#SBATCH -e /data01nfs/user/songq/log/trim.err
#SBATCH --no-requeue
#SBATCH -A cnl
export PATH=$PATH:/data01nfs/user/liupf/miniconda3/bin
source /data01nfs/user/songq/.bash_profile
source activate /data01nfs/user/liupf/miniconda3/envs/trimmomatic/
trimmomatic PE -threads 4  input.R1.fq.gz  input.R2.fq.gz  output_R1_trimmed.fq.gz  output_R1_trimmed_U.fq.gz  output_R2_trimmed.fq.gz  output_R2_trimmed_U.fq.gz ILLUMINACLIP:/data01nfs/user/liupf/common_files/TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
```

---

### Assemble

```bash
#SBATCH
source /data01nfs/user/liupf/miniconda3/bin/activate
conda activate megahit
megahit -1 input_R1.trimmed.fq.gz -2 input_R2.trimmed.fq.gz -t 60 --out-prefix --out-dir &> name.log
```

---
### >5k
```bash
#SBATCH 
export PATH=$PATH:/data01nfs/user/liupf/miniconda3/bin
source /data01nfs/user/songq/.bash_profile
source activate /datanode03/zhangxf/programs/mambaforge/envs/seqkit
while read id
do seqkit seq -m 5000 -j 8 <${id}_Raw_megahit_out> > ${id}_5k_contig.fa
done<id.list
```

---
### Genomic prediction

```bash
######VirSorter2######
#SBATCH
source /data01nfs/user/huangxy/programs/anaconda3/bin/activate
conda activate vs2
virsorter run -w <output >-i <input> --min-length 5000 --include-groups dsDNAphage,ssDNA,RNA,lavidaviridae,NCLDV -j 20 all

######VirFinder######
#SBATCH
source /data01nfs/user/liupf/miniconda3/bin/activative
conda activate virfinder
R
library("VirFinder")
predResult <- VF.pred("inputfile")
predResult[order(predResult$pvalue),]
predResult$qvalue <- VF.qvalue(predResult$pvalue)
predResult_sorter<-predResult[order(predResult$qvalue),]
write.table(predResult_sorter, file="output-prefix_virfinder.txt", quote=F, sep="\\t")

######geNomad######
#SBATCH
source /data01nfs/user/huangxy/programs/anaconda3/bin/activate
conda activate genomad
genomad end-to-end --min-score 0.7 --cleanup <input> <output> /data01nfs/user/liupf/oil_field_10kDNAvirus/genomad_db/genomad_db

######VIBRANT######
source /data01nfs/user/huangxy/programs/anaconda3/bin/activate
conda activate vibrant
VIBRANT_run.py -i input.fasta -t 10 -folder output_folder
conda deactivate

######DeepVirFinder######
#SBATCH
source /datanode03/huangxy/programs/anaconda3/bin/activate
conda activate deepvirfinder
python /datanode03/huangxy/programs/DeepVirFinder/dvf.py -i input_dir -o output_dir -l 1000 -c 10
conda deactivate
```
  
---
### Score-Screening
**VirSorter2**  
max_score >=0.9 == 1; 0.5-0.9 == 0.5; <0.5 == 0  
**geNomad**  
score >=0.8 == 1; 0.7-0.8 == 0.5; <0.7 == 0    

  
**Filter**  
VirSorter + geNomad >=1 ==> vContigs
### Check V
```bash
#### seqkit ####
#SBATCH

export PATH=$PATH:/data01nfs/user/liupf/miniconda3/bin
source /data01nfs/user/songq/.bash_profile
source activate /data01nfs/user/liupf/miniconda3/envs/seqkit
seqkit grep -f /datanode03/songq/seqname.id /datanode03/songq/5k_contigs/id_MV_MGHT_5K.fa -o /datanode03/songq/checkv_in/id.fa

#### checkv ####
#!/bin/bash
#SBATCH -J checkv
#SBATCH -p cn-long
#SBATCH -N 1
#SBATCH -c 60
#SBATCH -o /data01nfs/user/songq/log/checkv.out
#SBATCH -e /data01nfs/user/songq/log/checkv.err
#SBATCH --no-requeue
#SBATCH -A cnl

source /data01nfs/user/huangxy/programs/anaconda3/bin/activate
conda activate /data01nfs/user/huangxy/programs/anaconda3/envs/checkm
while read id
do
checkv end_to_end /datanode03/songq/checkv_in/${id} /datanode03/songq/checkv_out/${id} -t 30 -d /data01nfs/user/huangxy/database/checkv-db-v1.5
done<id.list
conda deactivate
```
### reCheckV
```Shell
#!/bin/bash
#SBATCH -J checkv
#SBATCH -p cn-long
#SBATCH -N 1
#SBATCH -c 60
#SBATCH -o /data01nfs/user/songq/log/checkv.out
#SBATCH -e /data01nfs/user/songq/log/checkv.err
#SBATCH --no-requeue
#SBATCH -A cnl
cd /datanode03/songq/recheckv_in
while read id
do
cat /datanode03/songq/checkv_out/${id}.fa/proviruses.fna /datanode03/songq/checkv_out/${id}.fa/viruses.fna > ${id}_virus_tmp.fasta
sed -i -e 's/ .*$//g' ${id}_virus_tmp.fasta
sed -i -e 's/_1$//g' ${id}_virus_tmp.fasta 
done<id.list

source /data01nfs/user/huangxy/programs/anaconda3/bin/activate
conda activate /data01nfs/user/huangxy/programs/anaconda3/envs/checkm

while read id
do
checkv end_to_end ${id}_virus_tmp.fasta /datanode03/songq/recheckv_out/${id} -t 30 -d /data01nfs/user/huangxy/database/checkv-db-v1.5
done<id.list
conda deactivate
```
---
Filter
---
### vOTU Clustering
Rapid genome clustering based on pairwise ANI

First, create a blast+ database:  
```Shell
makeblastdb -in <my_seqs.fna> -dbtype nucl -out <my_db> 
```

Next, use megablast from blast+ package to perform all-vs-all blastn of sequences:  
```Shell
blastn -query <my_seqs.fna> -db <my_db> -outfmt '6 std qlen slen' -max_target_seqs 10000 -o <my_blast.tsv> -num_threads 32
```

Next, calculate pairwise ANI by combining local alignments between sequence pairs:  
```Shell
anicalc.py -i <my_blast.tsv> -o <my_ani.tsv>
```

Finally, perform UCLUST-like clustering using the MIUVIG recommended-parameters (95% ANI + 85% AF):  
```Shell
aniclust.py --fna <my_seqs.fna> --ani <my_ani.tsv> --out <my_clusters.tsv> --min_ani 95 --min_tcov 85 --min_qcov 0
```