

### Trim

```bash
#!/bin/bash
#SBATCH -J trim
#SBATCH -p cn-long
#SBATCH -N 1
#SBATCH -c 60
#SBATCH -o /data01nfs/user/songq/log/trim.out
#SBATCH -e /data01nfs/user/songq/log/trim.err
#SBATCH --no-requeue
#SBATCH -A cnl
#SBATCH --mail-type=FAIL,END,BEGIN
#SBATCH --mail-user=2261518989@qq.com
R1=/R1_dir
R2=/R2_dir
R1_T=/output_R1_trimmed.fq.gz
R1_T_U=/output_R1_trimmed_U.fq.gz
R2_T=/output_R2_trimmed.fq.gz
R2_T_U=/output_R2_trimmed_U.fq.gz

conda activate /datanode03/songq/mambaforge/envs/trimmomatic
trimmomatic PE -threads 4  $R1 $R2 $R1_T $R1_T_U $R2_T $R2_T_U ILLUMINACLIP:/datanode03/liupf/common_files/TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
```

---

### Assemble
>7h+, out-dir中的最子端文件夹不能存在，megahit会自己创建。储存空间不足会报错 Exit Code:01
```bash
#SBATCH

conda activate megahit
megahit -1 input_R1.trimmed.fq.gz -2 input_R2.trimmed.fq.gz -t 60 --out-prefix prefix --out-dir dir &> name.log
```


### rename 
```shell
#SBATCH
source activate /data01nfs/user/liupf/miniconda3/envs/seqkit
while read id
do
seqkit replace -p "\s.+" ${id}.fa > ${id}_1.fa
#去掉空格后的东西
seqkit replace -p ^ -r <prefix_> ${id}_1.fa > ${id}_rename.fa
#加入样品id
done<id.list
conda deactivate
```

---
### >5k
```bash
#SBATCH 
source activate /datanode03/zhangxf/programs/mambaforge/envs/seqkit
while read id
do seqkit seq -m 5000 -j 8 <${id}_Raw_megahit_out> > ${id}_5k_contig.fa
done<id.list
```
`5k files cat > all_in_one_rename.fa`
<br>

### Genomic prediction 
```bash
######VirSorter2######
#SBATCH
source /datanode03/huangxy/programs/anaconda3/bin/activate
conda activate vs2
virsorter run -w <output >-i <input> --min-length 5000 --include-groups dsDNAphage,ssDNA,RNA,lavidaviridae,NCLDV -j 20 all

######VirFinder######
#########cancel
#SBATCH

conda activate virfinder
Rscript -e "
library("VirFinder")
predResult <- VF.pred('inputfile')
predResult[order(predResult$pvalue),]
predResult$qvalue <- VF.qvalue(predResult$pvalue)
predResult_sorter<-predResult[order(predResult$qvalue),]
write.table(predResult_sorter, file="output-prefix_virfinder.txt", quote=F, sep="\\t")
"
######geNomad######
#SBATCH
conda activate genomad
genomad end-to-end --min-score 0.7 --cleanup <input> <output> /datanode03/huangxy/database/genomad_db

######VIBRANT######
source /datanode03/huangxy/programs/anaconda3/bin/activate
conda activate vibrant
VIBRANT_run.py -i input.fasta -t 10 -folder output_folder
conda deactivate

######DeepVirFinder######
#SBATCH
source /datanode03/huangxy/programs/anaconda3/bin/activate
conda activate deepvirfinder
python /datanode03/huangxy/programs/DeepVirFinder/dvf.py -i input_dir -o output_dir -l 1000 -c 10
conda deactivate
```

---
### Score Filt
**VirSorter2**  
max_score >=0.9 == 1; 0.5-0.9 == 0.5; <0.5 == 0  
**geNomad**  
score >=0.8 == 1; 0.7-0.8 == 0.5; <0.7 == 0    
**DeepVirFinder**  
p-value<0.05 && score >= 0.9 == 1 ; 0.7-0.9 == 0.5 ; <0.7 == 0  
**VIBRANT**  
all == 1

*Filter*  
sum( All_Four_Score ) >=1 ==> vContigs

### R merge

```R
setwd("C:/Users/wudn2/OneDrive/桌面/2402")
dvf <- read.csv(file="dvf.csv",header=T)
vs2 <- read.csv(file="vs2.csv",header=T)
gnmd <- read.csv(file="gnmd.csv",header=T)
vbrt <- read.csv(file="vbrt.csv",header=T)

merge1 <- merge(vs2,vbrt,by.x="raw.id",by.y="raw.id",all.x=T,all.y=T)
merge2 <- merge(merge1,gnmd,by.x="raw.id",by.y="raw.id",all.x=T,all.y=T)
merge3 <- merge(merge2,dvf,by.x="raw.id",by.y="raw.id",all.x=T,all.y=T)

write.table(merge3,file="mergeall.txt",sep="\t",quote=F,col.names = T,row.names = F)
```
### Check V
```bash
#### seqkit ####
#SBATCH
source activate /data01nfs/user/liupf/miniconda3/envs/seqkit
seqkit grep -f score>1_id.txt all_in_one_rename.fa -o outputfile

#### checkv ####
#!/bin/bash
#SBATCH -J checkv
#SBATCH -p cn-long
#SBATCH -N 1
#SBATCH -c 60
#SBATCH -o /datanode03/songq/log/checkv.out
#SBATCH -e /datanode03/songq/log/checkv.err
#SBATCH --no-requeue
#SBATCH -A cnl
#SBATCH --mail-type=FAIL,END,BEGIN
#SBATCH --mail-user=your-email-address
source /datanode03/huangxy/programs/anaconda3/bin/activate
conda activate checkm

checkv end_to_end input output -t 30 -d /data01nfs/user/huangxy/database/checkv-db-v1.5

conda deactivate
```
### reCheckV
```Shell
#!/bin/bash
#SBATCH -J checkv
#SBATCH -p cn-long
#SBATCH -N 1
#SBATCH -c 60
#SBATCH -o /data01nfs/user/songq/log/checkv.out
#SBATCH -e /data01nfs/user/songq/log/checkv.err
#SBATCH --no-requeue
#SBATCH -A cnl
cd /datanode03/songq/recheckv_in
cat /datanode03/songq/checkv_out/proviruses.fna /datanode03/songq/checkv_out/viruses.fna > virus_tmp.fasta
sed -i -e 's/ .*$//g' virus_tmp.fasta
sed -i -e 's/_1$//g' virus_tmp.fasta 


source /data01nfs/user/huangxy/programs/anaconda3/bin/activate
conda activate /data01nfs/user/huangxy/programs/anaconda3/envs/checkm

checkv end_to_end virus_tmp.fasta /datanode03/songq/recheckv_out -t 30 -d /data01nfs/user/huangxy/database/checkv-db-v1.5

conda deactivate
```
---
### Filter  
reCheckV virus_gene >= 1

### vOTU Clustering
Rapid genome clustering based on pairwise ANI

First, create a blast+ database:  
```Shell
cat *.fa > blastdb.fa

source /data01nfs/apps/anaconda3/bin/activate
conda activate blast-2.5
makeblastdb -in blastdb.fa -dbtype nucl -out blastdb_0306/blastdb
conda activate
```

Next, use megablast from blast+ package to perform all-vs-all blastn of sequences:  
```Shell
blastn -query blastdb.fa -db blastdb_0306/blastdb -outfmt '6 std qlen slen' -max_target_seqs 10000 -out <my_blast.tsv> -num_threads 32
```

Next, calculate pairwise ANI by combining local alignments between sequence pairs:  
```Shell
python3 anicalc.py -i <my_blast.tsv> -o <my_ani.tsv>
```

Finally, perform UCLUST-like clustering using the MIUVIG recommended-parameters (95% ANI + 85% AF):  
```Shell
python3 aniclust.py --fna blastdb.fa --ani <my_ani.tsv> --out <my_clusters.tsv> --min_ani 95 --min_tcov 85 --min_qcov 0
```
### 得到 vOTU FASTA:去冗余后的病毒序列文件
```Shell
source activate /datanode03/zhangxf/programs/mambaforge/envs/seqkit
awk '{print $1}' my_clusters.tsv > my_cluster.list
seqkit grep -n -f my_cluster.list blastdb.fa -o vOTU.fa
conda deactivate
```

---
## Taxonomy

### blastn
```shell
source /data01nfs/apps/anaconda3/bin/activate
conda activate blast-2.5
blastn -task megablast -query vOTU.fna -db /datanode03/huangxy/database/IMG_VR_v4/IMG_VR_all/IMGVR_all_nucleotides -num_threads 16 -out /datanode03/songq/votu-blastn/HH_5KB_vrius1-2IMGVR.tsv -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore qlen slen" -perc_identity 90 -qcov_hsp_perc 75

conda activate /datanode03/zhangxf/programs/mambaforge/envs/blast
blastn -task megablast -query vOTU.fna -db /datanode03/huangxy/database/refvirus/viral.1.1.genomic -num_threads 16 -out /datanode03/songq/votu-blastn/HH_5KB_vrius1-2refvirus.tsv -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore qlen slen" -perc_identity 90 -qcov_hsp_perc 75
```
### vContact2
```shell
#step1 prodigal
source /apps/source/prodigal-2.6.3.sh
prodigal -p meta -q -m -i 23-24_vOTU_MG+MV_rm2303.fa -a 23-24_vOTU_MG+MV_rm2303.faa -d 23-24_vOTU_MG+MV_rm2303.fna -o 23-24_vOTU_MG+MV_rm2303.gff -f gff

#step2 vc2
conda activate vContact2

/datanode03/songq/mambaforge/envs/vContact2/MAVERICLab-vcontact2-c0413a6c92e8/bin/vcontact2_gene2genome -p 23-24_vOTU_MG+MV_rm2303.faa -o 23-24_vOTU_MG+MV_rm2303-vcontact2.csv -s Prodigal-FAA

/datanode03/songq/mambaforge/envs/vContact2/MAVERICLab-vcontact2-c0413a6c92e8/bin/vcontact2 --raw-proteins 23-24_vOTU_MG+MV_rm2303.faa --rel-mode Diamond --proteins-fp 23-24_vOTU_MG+MV_rm2303-vcontact2.csv --db 'ProkaryoticViralRefSeq201-Merged' --pcs-mode MCL --vcs-mode ClusterONE --c1-bin /datanode03/songq/mambaforge/envs/vContact2/MAVERICLab-vcontact2-c0413a6c92e8/bin/cluster_one-1.0.jar --output-dir ./  -t 30
```
### geNomad
```shell
...
```
### PhaGCN3
```shell
#!/bin/bash
#SBATCH -J PhaGCN
#SBATCH -p cn-long
#SBATCH -N 1
#SBATCH -c 60
#SBATCH -o /data01nfs/user/qinfs/log/pha.out
#SBATCH -e /data01nfs/user/qinfs/log/pha.err
#SBATCH --no-requeue
#SBATCH -A cnl
#SBATCH --mail-type=FAIL,END,BEGIN
#SBATCH --mail-user=2261518989@qq.com
cd /datanode03/zhujy/zhujy_datanode03/PhaGCN3
source activate /datanode03/zhaoy/.conda/envs/PhaGCN3/
python run_Speed_up.py --contigs ../DNA_virus/23-24_vOTU_MG+MV_rm2303.fa --outpath result_dir
```

### VPF-Class
```shell
# 登录Dell服务器

用登录其他账号的方法登录dell服务器。
username dell
key jyb-2021
Ip 172.21.231.31

# 新建文件夹
mkdir wenr

# 传输数据
#Dell无法访问/datanode03的数据，需要传输过来
#这一步需要重新登录自己的账号去传输

scp /datanode03/wenr/your_file dell@172.21.231.31:/home/dell/wenr

# 运行
#重新登录dell

cd wenr

#现在/home/dell/wenr/下应该只有输入文件。

mkdir your_out_put_dir

nohup /home/dell/huangxingyu/vpf-class-x86_64-linux@dd88a543f28eb339cf0dcb89f34479c84b3a8056  --data-index /home/dell/software/vpf-tools/vpf-class-data/index.yaml  -i your_input_file -o your_out_put_dir &

#回车运行。屏幕显示 【 nohup: 忽略输入并把输出追加到'nohup.out'】 之后就运行成功，可以进行 `ls` 等其他操作。
#log在nohup.out
#运行结束后scp将数据传输回/datanode03即可

#获得数据进行处理，例如famlily 以及genus这两个文件夹 membership_ratio和confidence_score进行条件筛选
具体是 membership_ratio> 0.5  confidence_score > 0.75
baltimore3.tsv 这个表格的筛选 membership_ratio> 0.2  confidence_score > 0.2
```
### ViralreCall 巨病毒注释
```shell
source /data01nfs/user/huangxy/programs/anaconda3/bin/activate
conda activate vibrant
cd /data01nfs/user/liupf/software_lpf/viralrecall
# 必须在这个目录下提交任务
mkdir songq songq_viralrecall-out 
cp dir/input.fa songq
nohup python ./viralrecall.py -i songq -p songq_viralrecall-out -t 6 -b &

##退出登录前可以通过jobs查看
#但是退出后，可以通过top -u liupf查看，退出该命令control + C 退出top页面
#输出主要看：.summary.tsv和.full_annot.tsv
#.full_annot.tsv主要看vog和pfam这两列，vog是一个专门的病毒数据库，这一列会给出病毒蛋白编号，pfam是解释这是什么蛋白
#.summary.tsv主要看marker基因，根据阈值是0.9 or 0.95来判断是巨病毒，也能通过巨病毒的marker基因来判断
```
### GV-Class 巨病毒注释
```shell
#要将输入文件切成一个序列/文件，放在输入文件夹下
source /datanode03/zhangxf/programs/mambaforge/bin/activate
conda activate gvclass
cd /datanode03/zhangxf/programs/mambaforge/envs/gvclass/gvclass
snakemake -j 30 --use-conda --config querydir="/datanode03/zhujy/zhujy_datanode03/DNA_virus/GVClass/extracted_sequences"
```
---
## Function
### DRAM-V
```shell
#Pre-DRAM

source activate /datanode03/huangxy/programs/anaconda3/envs/vs2
virsorter run --seqname-suffix-off --provirus-off  --keep-original-seq --viral-gene-enrich-off -w virsorter_prepare -i vOTU.fna --prep-for-dramv --min-length 0 --min-score 0  -j 16 all 

#DRAM-v

conda activate /datanode03/zhangxf/anaconda3/envs/DRAM
DRAM-v.py annotate -i virsorter_prepare/for-dramv/final-viral-combined-for-dramv.fa -v /datanode03/songq/virsorter_prepare/for-dramv/viral-affi-contigs-for-dramv.tab -o /datanode03/songq/dramv/dramv_result --threads 36
DRAM-v.py annotate -i vOTU.fna -o dramv/DRAMwithoutvs --threads 36

DRAM-v.py distill -i /datanode03/songq/dramv/dramv_result/annotations.tsv -o dramv-distill
DRAM-v.py distill -i /datanode03/songq/dramv/DRAMwithoutvs/annotations.tsv -o votu-distill

## 识别 AMG
DRAM-v 的 annotation.tsv 输出里 amg_flag 这一列仅包含 M、K、E这三个字母的。
```
---
## Abundance

```shell
#!/bin/bash
#SBATCH -J bowtie2
#SBATCH -p cn-long
#SBATCH -N 1
#SBATCH -c 50
#SBATCH -o /data01nfs/user/songq/log/bowtie2.out
#SBATCH -e /data01nfs/user/songq/log/bowtie2.err
#SBATCH --no-requeue
#SBATCH -A cnl
source /data01nfs/apps/anaconda3/bin/activate
conda activate bowtie2
while read id
R1trim="/datanode03/songq/Trimmed_out/${id}_MV/${id}.R1_trimmed.fq.gz"
R2trim="/datanode03/songq/Trimmed_out/${id}_MV/${id}.R2_trimmed.fq.gz"
bowtie2 -x bt2_ref/HH_vOTU_bowtie2.ref -1 ${R1trim} -2 ${R2trim} -S ${id}_HH_vOTU.sam --threads 36 --sensitive --no-unal
done <id.list

```
> The tutorial from ZXF ends here
---
## Host
### iPhop
```shell
#!/bin/bash
#SBATCH -J ip
#SBATCH -p cn-long
#SBATCH -N 1
#SBATCH -c 50
#SBATCH -o /data01nfs/user/songq/log/ip.out
#SBATCH -e /data01nfs/user/songq/log/ip.err
#SBATCH --no-requeue
#SBATCH -A cnl
#SBATCH --mail-type=FAIL,END,BEGIN
#SBATCH --mail-user=2261518989@qq.com
source activate /data01nfs/apps/anaconda3/envs/iphop-1.3.1
iphop predict --fa_file vOTU.fna --db_dir /data01nfs/biodatabase/iphop_db/Aug_2023_pub_rw --out_dir iphop_out
```

